### ISSUE TITLE
Error 2000 on mysql v3 throught tcp/ip

### ISSUE THREAD

**Issue by arkhenstone** [HeidiSQL/HeidiSQL#525]
## Steps to reproduce this issue

Setup a mysql server v3 or older, user, password, and domain name.

Fill in the host, user and password fields with Type : MariaDB or MySQL (TCP/IP)

Connect

## Current behavior

SQL Error (2000) : Unknown MySQL error

## Expected behavior

Connection should work

## Environment

Windows 10 and wine on the latest version of the setup at the time 10.1.0.5464 (64bit)
Mysql 3 on a windows machine, possibly XP.

## Details

Used to work in version 9 and older without any problem. Some sort of regression in the driver ?

**Comment by ansgarbecker** [comment]
Yes I guess it's the underlying libmariadb.dll and/or libmysql.dll which I updated for the 10.0 release.

There are [other clients as well with that issue](https://dba.stackexchange.com/questions/91712/whats-causing-this-error-2000-hy000-unknown-mysql-error), though I'm very unsure if the cause it's the same as for you. Maybe it's ok to break MySQL 3.x compatibility in HeidiSQL. MySQL itself burried it roughly 10 years ago, at least you don't get updates for it.

**Comment by arkhenstone** [comment]
It is possible to at least mention it in the patch notes if it was tested/known at the time it was deployed ?

As of now, anyone with old mysql database have no clues about the breaking of compatibility in this new version, and can update just to see it now set up a 2000 error.

**Comment by ansgarbecker** [comment]
Yes, I can add such a note on the download page.

**Comment by arkhenstone** [comment]
Thanks, I'll let you close this as you wish.

**Comment by ansgarbecker** [comment]
I just tested the current v12.16 and I can connect to my 23 years old mysql-3.23.58 instance. I only had to select `libmysql.dll` in the library dropdown. I got some SQL errors when I was trying to modify tables, but I can run queries just fine.

I don't recall right now when I introduced the library dropdown for such compatibility issues, but it could be after this issue has been reported.
